inductiva/fluids/scenarios/coastal_area/output.py

```diff
@@ -0,0 +1,242 @@
+            # again for consistency
+            transposed_data_values = data_values.transpose()
+
+            time_list.append(time_s)
+            data_list.append(transposed_data_values)
+
+    return (x_array, y_array), time_list, data_list
+
+
+def _render_quantity_data(x_array: np.ndarray,
+                          y_array: np.ndarray,
+                          time_list: Sequence[float],
+                          data_list: Sequence[np.ndarray],
+                          quantity: str,
+                          movie_path: str,
+                          fps: int = 10,
+                          cmap: Optional[str] = None,
+                          clim: Optional[Sequence[float]] = None):
+    """Renders temporal evolution of a quantity."""
+
+    with tempfile.TemporaryDirectory() as tmp_dir:
+
+        axes_limits = [
+            x_array.min(),
+            x_array.max(),
+            y_array.min(),
+            y_array.max(),
+        ]
+
+        if clim is None:
+            clim = [
+                np.min([np.min(data) for data in data_list]),
+                np.max([np.max(data) for data in data_list]),
+            ]
+
+        for idx, (time, data) in enumerate(zip(time_list, data_list)):
+            fig = plt.figure()
+            ax = fig.add_subplot()
+
+            im = ax.imshow(
+                np.transpose(data),
+                origin="lower",
+                cmap=cmap,
+                clim=clim,
+                extent=axes_limits,
+                aspect="equal",
+            )
+
+            plt.colorbar(
+                im,
+                ax=ax,
+                label=(f"{QUANTITY_PLOT_LABELS[quantity]} "
+                       f"[{QUANTITY_UNITS[quantity]}]"),
+            )
+
+            ax.set(
+                xlabel="x [m]",
+                ylabel="y [m]",
+                title=f"Time: {time:.2f} [s]",
+            )
+
+            fig.savefig(os.path.join(tmp_dir, f"{idx:06d}.png"))
+            plt.close(fig)
+
+        visualization.create_movie_from_frames(frames_dir=tmp_dir,
+                                               movie_path=movie_path,
+                                               fps=fps)
+
+
+class CoastalAreaOutput:
+    """Heat sink simulation output."""
+
+    def __init__(self, sim_output_path: Path):
+        """Initializes a `HeatSinkOutput` object.
+
+        Args:
+            sim_output_path: Path to simulation output files.
+        """
+
+        self.sim_output_path = sim_output_path
+
+    def render(
+        self,
+        quantity: Literal[
+            "water_level",
+            "velocity_x",
+            "velocity_y",
+            "velocity_magnitude",
+        ] = "water_level",
+        movie_path: Path = "movie.mp4",
+        fps: int = 10,
+        cmap: str = "viridis",
+        clim: Optional[Sequence[float]] = None,
+    ):
+        """Renders temporal evolution of a quantity.
```

Comment:
maybe say what type of quantity, like a physical quantity such as velocity. Quantity seems too general!